File: week10/weather_record.py
class WeatherRecord:

    def __init__(self, date, high, low):
        self.date = date
        self.high = high
        self.low  = low

    def getDate(self):
        return self.date

    def getHigh(self):
        return self.high

    def getLow(self):
        return self.low

    def __str__(self):
        return "On " + self.date + ", the high was " + str(self.high) + " and the low was " + str(self.low)

    
    
if __name__ == "__main__":

    sample = WeatherRecord("March 22, 2023", 30.5, 8.0)    
    anotherSample = WeatherRecord("The eleventeenth of Never", 100, -40.0);
    
    print(sample)
    print(anotherSample)

    print()

    print(sample.getDate())
    print(sample.getHigh())
    print(sample.getLow())
    
    

File: week10/weather_record.rb
class WeatherRecord

  # constructor
  def initialize(date, high, low)
    @date = date
    @high = high
    @low  = low
  end

  # convert the current object to a string
  def to_s()
    return "On #{@date}, the high was #{@high} and the low was #{@low}"
  end

  def getDate()
    return @date
  end

  def getHigh()
    return @high
  end

  def getLow()
    return @low
  end
  
end
    
    
sample = WeatherRecord.new("March 22, 2023", 30.5, 8.0)    
anotherSample = WeatherRecord.new("The eleventeenth of Never", 100, -40.0)
    
puts(sample)
puts(anotherSample)

puts()

puts(sample.getDate())
puts(sample.getHigh())
puts(sample.getLow())
    
    
